src/tbb/scheduler_common.h

```diff
@@ -439,6 +440,33 @@ struct suspend_point_type {
 #pragma warning( disable: 4324 )
 #endif
 
+class thread_reference_vertex: public d1::reference_vertex {
+public:
+    using d1::reference_vertex::reference_vertex;
+
+    void reserve(std::uint32_t delta = 1) override {
+        if (reserve_impl(delta) == 0) {
+            m_lifetime_ref_counter.fetch_add(1);
+        }
+    }
+
+    void release(std::uint32_t delta = 1) override {
+        if (release_impl(delta) == 0) {
+            destroy();
+        }
+    }
+
+    void destroy() {
+        auto ref = m_lifetime_ref_counter.fetch_sub(1);
+        if (ref == 1) {
+            this->~thread_reference_vertex();
+            cache_aligned_deallocate(this);
+        }
+    }
+private:
+    std::atomic<uint32_t> m_lifetime_ref_counter{1};
```

Comment:
Could you please add a comment that we reserve 1 in this reference counter in the constructor for the task_dispatcher that own the vertex map.

Reply:
Done